jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/resource/CombinedResource.java

```diff
@@ -26,36 +23,52 @@
 import java.util.List;
 import java.util.Objects;
 import java.util.stream.Collectors;
+import java.util.stream.Stream;
 
 import org.eclipse.jetty.util.URIUtil;
 
 /**
- * A collection of Resources.
- * Allows webapps to have multiple sources.
- * The first resource in the collection is the main resource.
- * If a resource is not found in the main resource, it looks it up in
- * the order the provided in the constructors
+ * Multiple resource directories presenteda as a single Resource.
  */
-public class ResourceCollection extends Resource
+public class CombinedResource extends Resource
 {
-    private final List<Resource> _resources;
+    /**
+     * Stream a resource to its component resources (if any).
+     * @param resource The resource to stream
+     * @return If the passed resource is a ResourceCollection, then it's contained
+     *         resources are streamed, otherwise the passed resource itself if streamed
+     */
+    public static Stream<Resource> stream(Resource resource)
```

Comment:
Wrong place for this.
Perhaps this should be in `Resource` or `Resources` ?

Reply:
Why not just have `.stream()` on the `Resource.stream()` interface instead?

Reply:
The style we have been adopting in jetty-12 is to avoid adding methods to interfaces if they can be implemented in a static like this.

However, this is not necessarily the correct place for this static.... and there is already the iterator API that does the same thing.... so I will review....

Reply:
moved to Resource